boot/bootutil/src/loader.c

```diff
@@ -1294,14 +1359,27 @@ boot_copy_image(struct boot_status *bs)
 #endif
     }
 
-    size = 0;
+    slot0_size = 0;
+    slot1_size = 0;
     last_sector_idx = 0;
+    /* not used in the swap routines, only here to match slot0's and
+     * slot1's sizes
+     */
+    last_idx_slot1 = 0;
     while (1) {
```

Comment:
I don't understand what happens in code below.

Reply:
Yeah, I don't think this is very intuitive. I added another comment just above the loop, should help in clarifying.